packages/koenig-lexical/src/components/KoenigCaptionEditor.jsx

```diff
@@ -9,7 +13,66 @@ const Placeholder = ({text = 'Type here'}) => {
     );
 };
 
+function CaptionPlugin({parentEditor}) {
+    const [editor] = useLexicalComposerContext();
+    const {setCaptionHasFocus, captionHasFocus} = useContext(CardContext);
+
+    // focus on caption editor when something is typed while card is selected
+    const handleKeyDown = useCallback((event) => {
+        // only if key is printable key, focus on editor
+        if (!captionHasFocus && event.key.length === 1 && !event.ctrlKey && !event.metaKey && !event.altKey) {
+            editor.focus();
+        }
+    }, [editor, captionHasFocus]);
+
+    React.useEffect(() => {
+        document.addEventListener('keydown', handleKeyDown);
+        return () => {
+            document.removeEventListener('keydown', handleKeyDown);
+        };
+    }, [handleKeyDown, editor]);
+
+    // handle focus/blur and enter key commands
+    React.useEffect(
+        () => {
+            return mergeRegister(
+                editor.registerCommand(
+                    FOCUS_COMMAND,
+                    () => {
+                        setCaptionHasFocus(true);
+                        return false;
+                    },
+                    COMMAND_PRIORITY_LOW
+                ),
+                editor.registerCommand(
+                    BLUR_COMMAND,
+                    () => {
+                        setCaptionHasFocus(false);
+                        editor.update(() => {
+                            $setSelection(null);
+                        });
+                        return false;
+                    },
+                    COMMAND_PRIORITY_LOW
+                ),
+                editor.registerCommand(
+                    KEY_ENTER_COMMAND,
+                    (event) => {
+                        parentEditor?.dispatchCommand(KEY_ENTER_COMMAND, event);
```

Comment:
I am not sure if this is the best way to handle the enter command here, and this does not work because of the logic we have in [here](https://github.com/TryGhost/Koenig/blob/f1b626a97866a532e42b2f72ef6610caf0b68478/packages/koenig-lexical/src/components/KoenigCardWrapper.jsx#L158-L160) to handle the Enter command in upstream editor. I am wondering, if it makes sense to add a new command on the main editor as `INSERT_NEW_PARAGRAPH` that can be dispatched from anywhere to get this behavior instead of relying on it like this and then using a hack upstream ?

Reply:
Could you use our `ExternalControlPlugin` here and add a command to that for inserting a paragraph?